tests/lib/rules/no-unsafe-negation.js

```diff
@@ -29,7 +48,16 @@ ruleTester.run("no-unsafe-negation", rule, {
         "a instanceof b",
         "a instanceof b === false",
         "!(a instanceof b)",
-        "(!a) instanceof b"
+        "(!a) instanceof b",
+        "if (! a < b) {}",
+        "while (! a > b) {}",
+        "foo = ! a <= b;",
+        "foo = ! a >= b;",
```

Comment:
Could we please also add a test with the option explicitly set to `false`, to be sure that the rule works well with the configuration. Maybe also one with an empty object (`options: [{}]`) to test the default value in `schema`.